controllers/dogu_manager.go

```diff
@@ -2,56 +2,41 @@ package controllers
 
 import (
 	"context"
-	"encoding/json"
 	"fmt"
 	cesappcore "github.com/cloudogu/cesapp-lib/core"
 	cesregistry "github.com/cloudogu/cesapp-lib/registry"
-	cesremote "github.com/cloudogu/cesapp-lib/remote"
 	"github.com/cloudogu/k8s-apply-lib/apply"
 	k8sv1 "github.com/cloudogu/k8s-dogu-operator/api/v1"
 	"github.com/cloudogu/k8s-dogu-operator/controllers/config"
-	"github.com/cloudogu/k8s-dogu-operator/controllers/dependency"
-	"github.com/cloudogu/k8s-dogu-operator/controllers/registry"
-	"github.com/cloudogu/k8s-dogu-operator/controllers/resource"
-	"github.com/cloudogu/k8s-dogu-operator/controllers/serviceaccount"
-	"github.com/go-logr/logr"
-	"sigs.k8s.io/yaml"
-
 	imagev1 "github.com/google/go-containerregistry/pkg/v1"
 	appsv1 "k8s.io/api/apps/v1"
 	corev1 "k8s.io/api/core/v1"
-	apierrors "k8s.io/apimachinery/pkg/api/errors"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/runtime"
-	"k8s.io/client-go/kubernetes"
-	ctrl "sigs.k8s.io/controller-runtime"
 	"sigs.k8s.io/controller-runtime/pkg/client"
-	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
 	"sigs.k8s.io/controller-runtime/pkg/log"
 )
 
-const finalizerName = "dogu-finalizer"
-const k8sDoguOperatorFieldManagerName = "k8s-dogu-operator"
-
 // NewManager is an alias mainly used for testing the main package
 var NewManager = NewDoguManager
 
 // DoguManager is a central unit in the process of handling dogu custom resources
 // The DoguManager creates, updates and deletes dogus
 type DoguManager struct {
 	client.Client
-	Scheme                *runtime.Scheme
-	ResourceGenerator     doguResourceGenerator
-	DoguRemoteRegistry    cesremote.Registry
-	DoguLocalRegistry     cesregistry.DoguRegistry
-	ImageRegistry         imageRegistry
-	DoguRegistrator       doguRegistrator
-	DependencyValidator   dependencyValidator
-	ServiceAccountCreator serviceAccountCreator
-	ServiceAccountRemover serviceAccountRemover
-	DoguSecretHandler     doguSecretHandler
-	FileExtractor         fileExtractor
-	Applier               applier
+	Scheme         *runtime.Scheme
+	InstallManager installManager
+	DeleteManager  deleteManager
+}
+
+type installManager interface {
+	// Install installs a dogu resource.
```

Comment:
IIRC the act of resource deletion is taken by the K8s core. I'd suggest that additionally all of our own resources will be deleted with the dogu being unregistered.

- [ ] hmmm

Reply:
Which resources stay after dogu deletion? (Except the dev configmap)